ncbi/src/main/java/org/reactome/release/ReactomeEvent.java

```diff
@@ -0,0 +1,159 @@
+package org.reactome.release;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.neo4j.driver.v1.Session;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+/**
+ * Class for describing Events (Pathways and Reaction Like Events) in Reactome.
+ * @author jweiser
+ */
+public class ReactomeEvent {
+	private static Map<Session, Map<Long, ReactomeEvent>> eventCache = new HashMap<>();
+	private static Logger logger = LogManager.getLogger();
+
+	private long dbId;
+	private String name;
+	private String stableIdentifier;
+
+	public ReactomeEvent(long dbId, String name, String stableIdentifier) {
+		this.dbId = dbId;
+		this.name = name;
+		this.stableIdentifier = stableIdentifier;
+	}
+
+	/**
+	 * Retrieves, from the graph database, the map of event database identifiers (both Pathways and
+	 * Reaction Like Events) to the ReactomeEvent object representing each event
+	 * @param graphDBSession Neo4J Driver Session object for querying the graph database
+	 * @return Map of Reactome Event database identifiers in Reactome to their ReactomeEvent objects
+	 */
+	public static Map<Long, ReactomeEvent> fetchReactomeEventMap(Session graphDBSession) {
+		if (eventCache.containsKey(graphDBSession)) {
+			return eventCache.get(graphDBSession);
+		}
+
+		logger.info("Computing Event map");
+
+		Map<Long, ReactomeEvent> eventMap = graphDBSession.run(
+			String.join(System.lineSeparator(),
+				"MATCH (e:Event)",
+				"RETURN e.dbId, e.displayName, e.stId"
+			)
+		)
+		.stream()
+		.collect(Collectors.toMap(
+			record -> record.get("e.dbId").asLong(),
+			record -> new ReactomeEvent(
+				record.get("e.dbId").asLong(),
+				record.get("e.displayName").asString(),
+				record.get("e.stId").asString()
+			)
+		));
+		eventCache.put(graphDBSession, eventMap);
+
+		logger.info("Finished computing Event map");
+
+		return eventMap;
+	}
+
+	/**
+	 * Retrieves the Reactome database identifier for the represented event
+	 * @return Reactome event database identifier
+	 */
+	public long getDbId() {
+		return this.dbId;
+	}
+
+	/**
+	 * Retrieves the Reactome display name for the represented event.  The name is standardized before being returned.
+	 * @return Reactome event display name
+	 */
+	public String getName() {
+		return fixName(this.name);
+	}
+
+	/**
+	 * Transforms Reactome event names that contain key words or phrases to standardized names
+	 * @param name Name to be considered for standardization
+	 * @return Standardized Reactome event display name if it contains pre-set words or phrases or the name passed as
+	 * a parameter, otherwise.
+	 */
+	private String fixName(String name) {
+		Map<String, String> patternToReplacement = new TreeMap<>();
+		patternToReplacement.put("amino acids", "Metabolism of nitrogenous molecules");
```

Comment:
Is there any worry about slightly off lettering or dashes, etc? This is safe as is, without a regex?

Reply:
I don't think so.  This has been in place for a long time and I think it might just be a failsafe in case old names are still present.

I may add checks to see if this is called and if it needs updating to be more flexible.